doc/contributing.mdwn

```diff
@@ -0,0 +1,207 @@
+  identified and agreed problem.
+* A patch must compile cleanly and pass project self-tests on all
+  target platforms.
+* A patch commit message must consist of a single short (less than 50
+  characters) line stating the a summary of the change, followed by a
+  blank line and then a description of the problem being solved and
+  its solution, or a reason for the change. Write more information,
+  not less, in the commit log.
+* Patches should be reviewed by at least one maintainer before being merged.
+
+Project maintainers should merge their own patches only when they have been
+approved by other maintainers, unless there is no response within a
+reasonable timeframe (roughly one week) or there is an urgent change
+to be done (e.g. security or data loss issue).
+
+As an exception to this rule, this specific document cannot be changed
+without the consensus of all administrators of the project.
+
+> Note: Those guidelines were inspired by the
+> [Collective Code Construct Contract][C4]. The document was found to
+> be a little too complex and hard to read and wasn't adopted in its
+> entirety. See this [discussion][] for more information.
+
+ [C4]: https://rfc.zeromq.org/spec:42/C4/
+ [discussion]: https://github.com/zeromq/rfc/issues?utf8=%E2%9C%93&q=author%3Aanarcat%20
+
+## Patch triage
+
+You can also review existing pull requests, by cloning the
+contributor's repository and testing it. If the tests do not pass
+(either locally or in Travis), if the patch is incomplete or otherwise
+does not respect the above guidelines, submit a review with "changes
+requested" with reasoning.
+
+# Issues and bug reports
+
+We want you to report issuess you find in the software. It is a
+recognized and important part of contributing to this project. All
+issues will be read and replied to politely and
+professionnally. Issues and bug reports should be filed on the
+[GitHub issue tracker][issues].
+
+## Issue triage
+
+Issue triage is a useful contribution as well. You can review the
+[issues][] in the GitHub project and, for each issue:
+
+-  try to reproduce the issue, if it is not reproducible, label it with
+   `help-wanted` and explain the steps taken to reproduce
+-  if information is missing, label it with `invalid` and request
+   specific information
+-  if the feature request is not within the scope of the project or
+   should be refused for other reasons, use the `wontfix` label and
+   close the issue
+-  mark feature requests with the `enhancement` label, bugs with
+   `bug`, duplicates with `duplicate` and so on...
+
+[issues]: https://github.com/linkcheck/linkchecker/issues
+
+Note that some of those operations are available only to project
+maintainers, see below for the different statuses.
+
+# Membership
+
+There are three levels of membership in the project, Administrator
+(also known as "Owner" in GitHub), Maintainer (also known as "Member")
+, or regular users (everyone with or without a GitHub account). Anyone
```

Comment:
The newline in front of a comma seems weird to me.

I'm still tempted to nitpick about "membership in the project".  I think there are two levels of membership, but an important point is that you don't have to be a member to contribute.